src/bootstrap/src/ferrocene/dist.rs

```diff
@@ -323,3 +325,107 @@ impl Step for TestOutcomes {
+        let src_channel = src_channel.trim();
+        let src_version = t!(fs::read_to_string("src/version"));
+        let src_version = src_version.trim();
+
+        // Perform validation on the contents of the version field, to avoid generating
+        // artifacts that will break the release process.
+        if ferrocene_channel == "rolling" && ferrocene_version != "rolling" {
+            panic!(
+                "error: ferrocene/version must be 'rolling' when ferrocene/ci/channel is 'rolling' but it was '{ferrocene_version}'"
+            );
+        }
+
+        let release_channel = match (src_channel, ferrocene_channel) {
+            ("nightly", "rolling") => "nightly".to_owned(),
+            ("beta", "rolling") => "pre-rolling".to_owned(),
+            ("stable", "rolling") => "rolling".to_owned(),
+            ("stable", ferrocene_channel @ ("beta" | "stable")) => {
+                let major_ferrocene = (|| {
+                    let mut version_components = ferrocene_version.split('.');
+                    let year = version_components.next()?;
+                    let month = version_components.next()?;
+                    let _patch = version_components.next()?;
+                    if version_components.next().is_none() {
+                        Some(format!("{year}.{month}"))
+                    } else {
+                        None
+                    }
+                })();
+                match major_ferrocene {
+                    Some(major_ferrocene) => format!("{ferrocene_channel}-{major_ferrocene}"),
+                    None => panic!(
+                        "invalid ferrocene/version, expected 'year.month.patch', got: {ferrocene_version}"
+                    ),
+                }
+            }
+            (rust, ferrocene) => panic!(
+                "error: unsupported channel configuration: rust '{rust}' and ferrocene '{ferrocene}'"
+            ),
+        };
+
+        let sha1_full = t!(std::process::Command::new("git").arg("rev-parse").arg("HEAD").output());
+        let sha1_full = t!(String::from_utf8(sha1_full.stdout));
+
+        let sha1_short = t!(std::process::Command::new("git")
+            .arg("rev-parse")
+            .arg("--short")
+            .arg("HEAD")
+            .output());
+        let sha1_short = t!(String::from_utf8(sha1_short.stdout));
+
+        // Whenever the contents of this JSON file change, even just adding new fields,
+        // make sure to increase the metadata version number and update publish-release
+        // accordingly. Note that new releases *won't* be made until publish-release and
+        // this use the same version number.
+        let data = format!(
+            r#"{{
+    "metadata_version": 3,
+    "rust_version": "{src_version}",
+    "rust_channel": "{src_channel}",
+    "ferrocene_version": "{ferrocene_version}",
+    "ferrocene_channel": "{ferrocene_channel}",
+    "release_channel": "{release_channel}",
```

Comment:
Taking offers for a better name here

Reply:
Just "channel" is also fine, that is what `publish-release` uses.